src/main/java/ca/sait/vezorla/controller/PaypalController.java

```diff
@@ -0,0 +1,99 @@
+package ca.sait.vezorla.controller;
+
+import ca.sait.vezorla.controller.util.CustomerClientUtil;
+import ca.sait.vezorla.exception.UnauthorizedException;
+import ca.sait.vezorla.model.Invoice;
+import ca.sait.vezorla.service.PaypalServices;
+import ca.sait.vezorla.service.UserServices;
+import ca.sait.vezorla.service.UserServicesImp;
+import com.paypal.api.payments.Links;
+import com.paypal.api.payments.Payment;
+import com.paypal.base.rest.PayPalRESTException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.EntityManager;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Controller for paypal payments
+ * JR is the number one
+ *
+ * Taken from: https://github.com/Java-Techie-jt/spring-boot-paypal-example/blob/master/src/main/java/com/javatechie/spring/paypal/api/PaypalController.java
+ */
+@Controller
+@CrossOrigin
+public class PaypalController {
+
+    public static final String SUCCESS_URL = "cart/payment/success";
+    public static final String CANCEL_URL = "cart/payment";
+    PaypalServices paypalServices;
+
+    private UserServices userServices;
+
+    public PaypalController(PaypalServices paypalServices, UserServices userServices){
```

Comment:
Please add `@AllArgsConstructor` and remove the constructor

Reply:
Done